src/compute/grid/track_sizing.rs

```diff
@@ -360,7 +360,13 @@ fn resolve_intrinsic_track_sizes(
             &get_track_size_estimate,
         );
 
-        let margin = item.margin.map(|m| m.resolve_or_zero(available_grid_space.width.into_option())).sum_axes();
+        let margin = Rect {
+            left: item.margin.left.resolve_or_zero(Some(0.0)),
```

Comment:
I'm curious if it would be possible to rewrite our resolve-systems so that we could write it something like this:
```rs
            left: item.margin.left.resolve_or(zero()),
```
And that it would figure out what type the zero() would need to output. 

Not sure if it would work, or if it did, if it would improve readability of the codebase at all.

Reply:
I think that wouldn't actually do quite the same thing. It's actually a coincidence that the argument is zero here. In general the `Some(0.0)` will be the actual size of the parent node. It's only zero in this case because in this specific scenario horizontal percentage margins *always* resolve to zero (because otherwise there would be a cyclic dependency).

What this function is doing is:
- Returning `Points` values
- Resolving `Percent` values against the passed in argument (if it's `Some`)
- Else returning `0.0`